### ISSUE TITLE
`OpenXRInterface.get_transform_for_view` crashes Godot

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#68515]
### Godot version

4.0 5683aa2498e5ae65d27a772fbf6b8b9fd93e0b28

### System information

Ubuntu 22.04 headless

### Issue description

```
extends Node
func _process(delta):

	var temp_variable6154 = OpenXRInterface.new()
	temp_variable6154.get_transform_for_view(72, Transform3D(Basis(Vector3(-23.8213367462158, 45.6554183959961, 6.31532669067383), Vector3(-25.8811588287354, 49.5911827087402, 70.4532165527344), Vector3(7.34195709228516, 40.610969543457, -42.0984153747559)), Vector3(-0.06058242917061, 0.69352751970291, -0.7178783416748)))

```
crashes Godot with this backtrace (problably checking index should fix issue)
```
 modules/openxr/openxr_interface.cpp:676:32: runtime error: index 72 out of bounds for type 'Transform3D [2]'
=================================================================
==30016==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6170000dd150 at pc 0x55ea87179d10 bp 0x7ffcb9ff8560 sp 0x7ffcb9ff8550
READ of size 48 at 0x6170000dd150 thread T0
    #0 0x55ea87179d0f in OpenXRInterface::get_transform_for_view(unsigned int, Transform3D const&) modules/openxr/openxr_interface.cpp:676
    #1 0x55ea986b1fe1 in void call_with_variant_args_ret_helper<__UnexistingClass, Transform3D, unsigned int, Transform3D const&, 0ul, 1ul>(__UnexistingClass*, Transform3D (__UnexistingClass::*)(unsigned int, Transform3D const&), Variant const**, Variant&, Callable::CallError&, IndexSequence<0ul, 1ul>) core/variant/binder_common.h:680
    #2 0x55ea986ad943 in void call_with_variant_args_ret_dv<__UnexistingClass, Transform3D, unsigned int, Transform3D const&>(__UnexistingClass*, Transform3D (__UnexistingClass::*)(unsigned int, Transform3D const&), Variant const**, int, Variant&, Callable::CallError&, Vector<Variant> const&) core/variant/binder_common.h:493
    #3 0x55ea986aa4bb in MethodBindTR<Transform3D, unsigned int, Transform3D const&>::call(Object*, Variant const**, int, Callable::CallError&) core/object/method_bind.h:465
    #4 0x55ea9d58c410 in Object::callp(StringName const&, Variant const**, int, Callable::CallError&) core/object/object.cpp:733
    #5 0x55ea9ca4cd9a in Variant::callp(StringName const&, Variant const**, int, Variant&, Callable::CallError&) core/variant/variant_call.cpp:1048
    #6 0x55ea85036472 in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_vm.cpp:1555
    #7 0x55ea84a2d0f0 in GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) modules/gdscript/gdscript.cpp:1634
    #8 0x55ea92314436 in bool Node::_gdvirtual__process_call<false>(double) scene/main/node.h:238
    #9 0x55ea9227ad16 in Node::_notification(int) scene/main/node.cpp:56
    #10 0x55ea83584757 in Node::_notificationv(int, bool) scene/main/node.h:45
    #11 0x55ea9d58d964 in Object::notification(int, bool) core/object/object.cpp:790
    #12 0x55ea923d9ceb in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:868
    #13 0x55ea923cc4a7 in SceneTree::process(double) scene/main/scene_tree.cpp:466
    #14 0x55ea82e28d38 in Main::iteration() main/main.cpp:3163
    #15 0x55ea82c562ed in OS_LinuxBSD::run() platform/linuxbsd/os_linuxbsd.cpp:785
    #16 0x55ea82c3337b in main platform/linuxbsd/godot_linuxbsd.cpp:73
    #17 0x7f82807eed8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)
    #18 0x7f82807eee3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e3f)
    #19 0x55ea82c32d34 in _start (/home/runner/work/Qarminer/Qarminer/godot.linuxbsd.editor.dev.x86_64.san+0x392a0d34)

0x6170000dd150 is located 3312 bytes to the right of 736-byte region [0x6170000dc180,0x6170000dc460)
allocated by thread T0 here:
    #0 0x7f82813da867 in __interceptor_malloc ../../../../src/libsanitizer/asan/asan_malloc_linux.cpp:145
    #1 0x55ea9bd7b869 in Memory::alloc_static(unsigned long, bool) core/os/memory.cpp:75
    #2 0x55ea9bd7b77a in operator new(unsigned long, char const*) core/os/memory.cpp:40
    #3 0x55ea8701a38f in Object* ClassDB::creator<OpenXRInterface>() core/object/class_db.h:140
    #4 0x55ea9d4f246d in ClassDB::instantiate(StringName const&) core/object/class_db.cpp:339
    #5 0x55ea849ef5cf in GDScriptNativeClass::instantiate() modules/gdscript/gdscript.cpp:92
    #6 0x55ea849eef87 in GDScriptNativeClass::_new() modules/gdscript/gdscript.cpp:80
    #7 0x55ea84b26df3 in void call_with_variant_args_ret_helper<__UnexistingClass, Variant>(__UnexistingClass*, Variant (__UnexistingClass::*)(), Variant const**, Variant&, Callable::CallError&, IndexSequence<>) core/variant/binder_common.h:680
    #8 0x55ea84b25e84 in void call_with_variant_args_ret_dv<__UnexistingClass, Variant>(__UnexistingClass*, Variant (__UnexistingClass::*)(), Variant const**, int, Variant&, Callable::CallError&, Vector<Variant> const&) core/variant/binder_common.h:493
    #9 0x55ea84b24a13 in MethodBindTR<Variant>::call(Object*, Variant const**, int, Callable::CallError&) core/object/method_bind.h:465
    #10 0x55ea85038a6d in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_vm.cpp:1644
    #11 0x55ea84a2d0f0 in GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) modules/gdscript/gdscript.cpp:1634
    #12 0x55ea92314436 in bool Node::_gdvirtual__process_call<false>(double) scene/main/node.h:238
    #13 0x55ea9227ad16 in Node::_notification(int) scene/main/node.cpp:56
    #14 0x55ea83584757 in Node::_notificationv(int, bool) scene/main/node.h:45
    #15 0x55ea9d58d964 in Object::notification(int, bool) core/object/object.cpp:790
    #16 0x55ea923d9ceb in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:868
    #17 0x55ea923cc4a7 in SceneTree::process(double) scene/main/scene_tree.cpp:466
    #18 0x55ea82e28d38 in Main::iteration() main/main.cpp:3163
    #19 0x55ea82c562ed in OS_LinuxBSD::run() platform/linuxbsd/os_linuxbsd.cpp:785
    #20 0x55ea82c3337b in main platform/linuxbsd/godot_linuxbsd.cpp:73
    #21 0x7f82807eed8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)
```

### Steps to reproduce

above

**Comment by cheece** [comment]
Since it might affect other implementations of XRInterface I wonder if it would be better to replace the exposed method by a wrapper that does the checking in XRInterface itself.

**Comment by akien-mga** [comment]
Fixed by #68519.
